File: server/src/main/java/br/app/pregsrateio/usuario/data/Usuario.java
package br.app.pregsrateio.usuario.data;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.bson.types.ObjectId;
import lombok.*;
import java.time.Instant;

// TODO: Adicionar campos de auditoria
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "usuarios")
public class Usuario {
    @Id
    private ObjectId id;

    private String authSub;

    private String nome;

    private String sobrenome;

    private String email;

    private String numeroCelular;

    private String fotoUrl;

    @Builder.Default
    private boolean cadastroCompleto = true;
}

File: server/src/main/kotlin/br/dev/brunorsch/pregsrateio/usuario/model/Usuario.kt
package br.dev.brunorsch.pregsrateio.usuario.model

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import org.bson.types.ObjectId

@MappedEntity
data class Usuario(
    @field:Id
    @GeneratedValue
    val id: ObjectId? = null,
    val authSub: String,
    val nome: String,
    val sobrenome: String,
    val email: String,
    val whatsApp: String,
    val fotoUrl: String? = null,
    val isCadastroCompleto: Boolean = true,
)